patches/tModLoader/Terraria/ModLoader/GlobalNPC.cs

```diff
@@ -85,14 +85,22 @@ public virtual void ModifyTypeName(NPC npc, ref string typeName)
 
 	/// <summary>
 	/// Allows you to modify the bounding box for hovering over the given NPC (affects things like whether or not its name is displayed).
+	/// <br/>This hook is not called if <see cref="NPC.ShowNameOnHover"/> is set to <see langword="false">.
 	/// </summary>
 	/// <param name="npc">The NPC in question.</param>
 	/// <param name="boundingBox">The bounding box used for determining whether or not the NPC counts as being hovered over.</param>
 	public virtual void ModifyHoverBoundingBox(NPC npc, ref Rectangle boundingBox)
 	{
 	}
 
-	public virtual void OnHoverBoundingBox(NPC npc, Rectangle boundingBox, bool mouseIntersects, bool smartCursorInteraction)
+	/// <summary>
+	/// Gets called for hovering over the given NPC (things like whether or not its name is displayed).
+	/// <br/>This hook is not called if <see cref="NPC.ShowNameOnHover"/> is set to <see langword="false">.
+	/// </summary>
+	/// <param name="npc">The NPC in question.</param>
+	/// <param name="boundingBox">The bounding box used for determining whether or not the NPC counts as being hovered over.</param>
+	/// <param name="mouseIntersects">Whether or not mouse is actually hovering over the NPC's bounding box.</param>
```

Comment:
```
Called when the the NPC is selected via smart interact, or when the mouse is hovering over it.
<br/>Use to ...
<br/>
<br/>Only works on NPCs where <see cref="NPC.ShowNameOnHover"/> returns <see langword="true">
```

Reply:
wouldn'tit be better to have "only works on npcs where" in remarks